system/lib/sbrk.c

```diff
@@ -27,30 +27,57 @@ extern size_t emscripten_get_heap_size(void);
 #endif
 
 void *sbrk(intptr_t increment) {
-  intptr_t* sbrk_ptr = emscripten_get_sbrk_ptr();
-  intptr_t old_brk = *sbrk_ptr;
-  // TODO: overflow checks
-  intptr_t new_brk = old_brk + increment;
-#if __wasm__
-  uintptr_t old_size = __builtin_wasm_memory_size(0) * WASM_PAGE_SIZE;
+#if __EMSCRIPTEN_PTHREADS__
+  // Our default dlmalloc uses locks around each malloc/free, so no additional
+  // work is necessary to keep things threadsafe, but we also make sure sbrk
+  // itself is threadsafe so alternative allocators work. We do that by looping
+  // and retrying if we hit interference with another thread.
+  while (1) {
+#endif // __EMSCRIPTEN_PTHREADS__
+
+    intptr_t* sbrk_ptr = emscripten_get_sbrk_ptr();
```

Comment:
maybe this is an existing issue (and it's definitely a nit), but it might be clearer to name this `emscripten_get_brk` or `get_brk_ptr`, since it's `brk` that sets the location of the program break and `sbrk` is just a delta.

Reply:
Hmm, yeah, maybe `get_brk_ptr` would be better. I think it's ok as it is though, since it's the ptr used by sbrk/brk? If you feel strongly we can rename it (will take work in binaryen too).

Reply:
If this is essentially a private API, then it may not be worth it for that. And if it's a longstanding public API, then maybe we don't want to break it :)

Reply:
It's a private api basically, that communicates between emscripten and binaryen, but nothing else should be aware of it.